kitsune/kpi/static/kpi/js/components/Chart.es6.js

```diff
@@ -0,0 +1,182 @@
+      buckets: 10,
+      legendElementWidth: 95,
+      data: [],
+      dom: {
+        graphContainer: $container.find('.graph').get()[0]
+      }
+    };
+
+    $.extend(true, this, defaults, options);
+
+    this.init();
+  }
+
+  // Render whatever pieces of the chart we can while waiting for data
+  preRender() {
+    // draw the container svg for the chart
+    this.dom.svg = d3.select(this.dom.graphContainer).append('svg')
+      .attr('width', this.width + this.margin.left + this.margin.right)
+      .attr('height', this.height + this.margin.top + this.margin.bottom)
+      .append('g')
+        .attr('transform', 'translate(' + this.margin.left + ',' + this.margin.top + ')');
+
+    this.dom.svg.append('g')
+      .attr('class', 'data');
+
+    this.setupAxis('xAxis');
+    this.setupLegend();
+  }
+
+  setupAxis(axis) {
+    let self = this;
+    let axisGroup = self.dom.svg.append('g')
+      .attr('class', axis);
+
+    axisGroup.selectAll()
+      .data(self.axes[axis].labels)
+        .enter().append('text')
+        .text(function(d,i) { return d; })
+        .attr('x', function(d, i) {
+          return self.axes.getPosition('x', axis, i) + self.axes[axis].labelOffsets.x;
+        })
+        .attr('y', function(d, i) {
+          return self.axes.getPosition('y', axis, i) + self.axes[axis].labelOffsets.y;
+        })
+  }
+
+  setupLegend() {
+    let self = this;
+    let legendData = self.chartColors;
+    let legendYPosition = (self.grid.rows * self.gridSize/2) + self.gridSize;
+    let legendXPositions = i => {
+      return (self.legendElementWidth * i) - self.gridSize;
+    }
+
+    let legend = self.dom.svg.append('g')
+      .attr('class', 'legend');
+
+    legend.selectAll('rect')
+      .data(legendData, function(d) { return d; })
+        .enter().append('rect')
+        .attr('x', function(d, i) { return legendXPositions(i); })
+        .attr('y', legendYPosition)
+        .attr('width', self.legendElementWidth)
+        .attr('height', self.gridSize / 3.6)
+        .style('fill', function(d, i) { return self.chartColors[i]; });
+
+    legend.selectAll('text')
+      .data(legendData, function(d) { return d; })
+        .enter().append('text')
+        .text(function(d, i) { return '≥ ' + Math.round(i / self.buckets * 100) + '%'; })
+        .attr('x', function(d, i) { return legendXPositions(i) + 7; })
+        .attr('y', legendYPosition + 14);
+  }
+
+  init() {
+    this.preRender();
+  }
+
+  setupGrid(filteredData, filter) {
+    let self = this;
+    let kindFilter = filter;
+
+    self.dom.cohorts = self.dom.svg.select('.data').selectAll('g')
+      .data(filteredData);
+
+    self.dom.cohorts.enter().append('g');
+
+    self.dom.cohorts
+      .attr('width', self.width)
+      .attr('height', self.gridSize/2)
+      .attr('x', 0)
+      .attr('y', function(d, i) {
+        return i * self.gridSize/2;
+      })
+      .attr('class', function(d, i) {
+        return 'cohort-group ' + kindFilter;
+      });
```

Comment:
This can just be a constant, right? `.attr('class', 'cohort-group ' + kindfilter)`?